modules/javafx.graphics/src/main/java/javafx/print/Paper.java

```diff
@@ -78,9 +78,9 @@ public final String getName() {
      */
     private double getSizeInPoints(double dim) {
         switch (units) {
-        case POINT : return (int)(dim+0.5);
-        case INCH  : return (int)((dim * 72) + 0.5);
-        case MM    : return (int)(((dim * 72) / 25.4) + 0.5);
+            case POINT : return dim;
+            case INCH  : return dim * 72;
+            case MM    : return (dim * 72) / 25.4;
```

Comment:
yeah,  I also don't quite understand why we have the rounding (I mean, floor'ing) here.
I would vote to remove the typecast as the OP proposes.

But there is more:
1. why in the world the constructor is not public?  what if I need to print a Commercial 8-5/8 envelope? 
see https://www.papersizes.org/us-envelope-sizes.htm

2. what is going on in hashCode??
```
    public final int hashCode() {
        return (int)width+((int)height<<16)+units.hashCode();
    }
```

Reply:
> yeah, I also don't quite understand why we have the rounding (I mean, floor'ing) here. I would vote to remove the typecast as the OP proposes.

Phil has some concerns that will need to be looked into.

> But there is more:
> 
> 1. why in the world the constructor is not public?  what if I need to print a Commercial 8-5/8 envelope?

That's the wrong question to ask, and this PR is the wrong place to ask it. If I could reformulate your question, I might rather state it as "would it be a useful enhancement to allow Paper to be customizable?". Maybe (or maybe not), but as with any other new feature, it needs proper motivation, etc.

> 2. what is going on in hashCode??
> 
> ```
>     public final int hashCode() {
>         return (int)width+((int)height<<16)+units.hashCode();
>     }
> ```

This is also unrelated to this PR. It may or may not be a great `hashCode` implementation, but It meets the contract of hasCode w.r.t. equals. And if it didn't, it would need a separate bug.

Reply:
> > yeah, I also don't quite understand why we have the rounding (I mean, floor'ing) here. I would vote to remove the typecast as the OP proposes.
> 
> Phil has some concerns that will need to be looked into.

To follow-on, it seems that there are a few options.

1. Pass back the unmodified value. The concern that Phil raised is that this could lead to floating-point error. Of course the current code that truncates could also lead to errors. If the value ends up as, say, 10.9999 inches, we would want to return 11 rather than 10 even if we did want to limit it to integers.
2. Round the value to whole integers. This doesn't really solve the problem that the PR is trying to solve
3. Round to some number of decimal places, e.g., round to the nearest 0.01 or 0.001 inches.
4. Something else.

Option 3 seems like it might be a reasonable solution, but I'd want Phil to weigh in.

Reply:
what kind of FP error could option 1 lead to?

I think rounding belongs to the UI, to the panel which displays the sizes with precision given by the requirements.  So this method, I think, should return the unmodified value.

Looking at AWT Paper, I see no rounding there.  Where did it come from to FX?

PS. my other comments are indeed unrelated to this PR.  sorry.

Reply:
Some observations on the paper class:

    /**
     * Specifies the North American legal size, 8.5 inches by 14 inches.
     */
    public static final Paper LEGAL = new Paper("Legal", 8.4, 14, INCH);


The comment and actual size contradict each other. 8.4 inches would not result in an integer number of points (604.8) and looking up the legal paper size confirms it should be 8.5 inches

    /**
     * Specifies the Monarch envelope size, 3.87 inch by 7.5 inch.
     */
    public static final Paper MONARCH_ENVELOPE = new Paper("Monarch Envelope", 3.87, 7.5, INCH);

This seems incorrect, the size is 3-7/8” x 7-1/2”, which would be `3.875` x `7.5` -- the difference in this case is small enough that it will be rounded to the same value in points.

Reply:
About the rounding; all these values are fixed values and could have been entered in points as constants directly.

Also realize that all papers measured in inches can be expressed exactly in points (after the above errors are fixed) so no rounding is needed for those at all; the same is not true for the papers specified in millimeters.

Since the papers in inches never should need any rounding, the rounding can therefore be removed for those.  If there is a worry that the multiplication (* 72) may cause floating point errors, then I suggest to specify all papers in points immediately, like this:

    new Paper("Letter", 8.5 * 72, 11 * 72, POINT);

As for the papers specified in millimeters, these can't be expressed in whole points, as a millimeter is equivalent to `2,83465` points.  Rounding them to points will require some guess work to find out what the actual size in millimeters was, for example, when it says 3 points it must have been 1 mm, while 1, 2, 4 or 5 points has no direct mm equivalent.

Reply:
It is also very odd that `Paper` would expose the name (like `A4`), but not the unit (`MM`) and the exact sizes for those.  Representing these in a UI therefore becomes terribly complicated; if I want to show the user, I can show them the name, but then when I want to display its size, I have to show it in points (a useless measurement, for users, for both papers specific in inches and mm).

Reply:
Typically paper sizes are displayed in whatever units are appropriate for the locale. On my Mac all paper sizes are displayed in whole millimeters if the locale is United Kingdom and inches (to 1/100 of an inch) if the locale is United States.